src/test/groovy/com/netflix/nebula/lint/rule/dependency/DeprecatedDependencyConfigurationRuleSpec.groovy

```diff
@@ -0,0 +1,272 @@
+import spock.lang.Subject
+import spock.lang.Unroll
+
+@Unroll
+@Subject(DeprecatedDependencyConfigurationRule)
+class DeprecatedDependencyConfigurationRuleSpec extends TestKitSpecification {
+
+    def 'Replaces deprecated configurations - #configuration for #replacementConfiguration'() {
+        buildFile << """
+            plugins {
+                id 'nebula.lint'
+                id 'java'
+            }
+
+            gradleLint.rules = ['deprecated-dependency-configuration']
+
+            repositories {
+                mavenCentral()
+            }
+            
+            dependencies {
+                $configuration 'com.google.guava:guava:19.0'
+            }
+        """
+
+        when:
+        def result = runTasksSuccessfully('fixGradleLint', '--warning-mode=none')
+
+        then:
+        def buildGradle = new File(projectDir, 'build.gradle')
+        buildGradle.text.trim() == """
+            plugins {
+                id 'nebula.lint'
+                id 'java'
+            }
+
+            gradleLint.rules = ['deprecated-dependency-configuration']
+
+            repositories {
+                mavenCentral()
+            }
+            
+            dependencies {
+                $replacementConfiguration 'com.google.guava:guava:19.0'
+            }
+        """.trim()
+
+        where:
+        configuration | replacementConfiguration
+        "compile"     | "implementation"
+        "testCompile" | "testImplementation"
+        "runtime"     | "runtimeOnly"
+    }
+
+    def 'Replaces deprecated configurations - multi project - project dependency - #configuration for #replacementConfiguration'() {
+        def sub1 = addSubproject('sub1', """
+            repositories {
+                mavenCentral()
+            }
+            
+            dependencies {
+                $configuration 'com.google.guava:guava:19.0'
+                $configuration project(':sub2')
+            }
+
+            def x = "test"
+            """.stripIndent())
+
+        createJavaSourceFile(sub1, 'public class Sub1 {}')
+
+        def sub2 = addSubproject('sub2', """            
+            repositories {
+                mavenCentral()
+            }
+            
+            dependencies {
+                $configuration 'com.google.guava:guava:19.0'
+            }
+            """.stripIndent())
+
+        createJavaSourceFile(sub2, 'public class Sub2 {}')
+
+        buildFile << """
+            plugins {
+                id 'nebula.lint'
+            }
+            
+            subprojects {
+                apply plugin: 'java'
+                apply plugin: 'nebula.lint'
+            }
+              
+            allprojects {
+                  gradleLint.rules = ['deprecated-dependency-configuration']
+            }
+
+        """
+
+        when:
+        def result = runTasksSuccessfully('fixGradleLint', '--warning-mode=none')
+
+        then:
+        def buildGradle = new File(projectDir, 'sub1/build.gradle')
```

Comment:
I think it would be clearer for this variable name to indicate it is the sub1's build file

Reply:
done 👍